metricsaggregator/src/main/java/io/dockstore/metricsaggregator/helper/AggregationHelper.java

```diff
@@ -0,0 +1,125 @@
+import io.dockstore.openapi.client.model.ExecutionTimeMetric;
+import io.dockstore.openapi.client.model.MemoryMetric;
+import io.dockstore.openapi.client.model.Metrics;
+import io.dockstore.webservice.core.metrics.MemoryStatisticMetric;
+import java.time.Duration;
+import java.time.format.DateTimeParseException;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public final class AggregationHelper {
+    private static final Logger LOG = LoggerFactory.getLogger(AggregationHelper.class);
+
+    private AggregationHelper() {}
+
+    public static Optional<Metrics> getAggregatedMetrics(List<Execution> executions) {
+        Optional<ExecutionStatusMetric> aggregatedExecutionStatus = getAggregatedExecutionStatus(executions);
+        if (getAggregatedExecutionStatus(executions).isPresent()) {
+            Metrics aggregatedMetrics = new Metrics();
+            aggregatedMetrics.setExecutionStatusCount(aggregatedExecutionStatus.get());
+            getAggregatedExecutionTime(executions).ifPresent(aggregatedMetrics::setExecutionTime);
+            getAggregatedCpu(executions).ifPresent(aggregatedMetrics::setCpu);
+            getAggregatedMemory(executions).ifPresent(aggregatedMetrics::setMemory);
+            return Optional.of(aggregatedMetrics);
+        }
+        return Optional.empty();
+    }
+
+    public static Optional<ExecutionStatusMetric> getAggregatedExecutionStatus(List<Execution> executions) {
+        Map<String, Integer> statusCount = executions.stream()
+                .map(execution -> execution.getExecutionStatus().toString())
+                .collect(groupingBy(Function.identity(), Collectors.reducing(0, e -> 1, Integer::sum)));
+        // This shouldn't happen because all executions should have an execution status, but check anyway
+        if (statusCount.isEmpty()) {
+            return Optional.empty();
+        }
+        return Optional.of(new ExecutionStatusMetric().count(statusCount));
+    }
+
+    public static Optional<ExecutionTimeMetric> getAggregatedExecutionTime(List<Execution> executions) {
+        List<String> executionTimes = executions.stream()
+                .map(Execution::getExecutionTime)
+                .filter(Objects::nonNull)
+                .toList();
+        List<Double> executionTimesInSeconds = executionTimes.stream()
+                .map(executionTime -> {
+                    // Convert executionTime in ISO 8601 duration format to seconds
+                    try {
+                        return Long.valueOf(Duration.parse(executionTime).toSeconds()).doubleValue();
+                    } catch (DateTimeParseException e) {
+                        LOG.error("Could not parse Duration from {}", executionTime);
+                        return null;
```

Comment:
Do you need to filter out the nulls after the `map`? It's not clear to me if you do or don't. If you don't need to filter them, perhaps a comment on why not.

Reply:
Good catch, yes I need to filter out the nulls